### PR TITLE
Update SPA and DNN Dependencies, Add Unpublished Pages Toggle, Refactor Symbols Packaging

### LINKED ISSUE #120: Filter by Unpublished Pages
## Sponsorship  

If this request requires additional support (e.g., such as direct email/phone/meeting/development), I have the following interest in helping to [sponsor the effot via GitHub Sponsors](https://github.com/sponsors/UpendoVentures):    

- [ ] None, please continute to work for me for free :P  
- [ ] Absolutely, I get value out of this!  
- [ ] Maybe later  
- [x] I'm already a sponsor... Woot!  

## Is your feature request related to a problem?
When viewing the pages on a site with a lot of pages, one may need to view which page(s) are unpublished. Right now, this is shown, but there isn't a way to see only those pages.  

## Describe the solution you'd like
Remove the word "filter" from the existing Metadata toggle.  
Add another toggle next to the Metadata toggle, labeled "Unpublished" and it's default state is `Off`.
Only one of the toggles can be turned on at a time.  If one is switched, the other is turned off (if necessary).  

## Describe alternatives you've considered
None at this time. 

## Additional context
None.

### PR DESCRIPTION
**Pull request by ArielBlanco1990**
## Related to Issue
Fixes #120, #58, #57, #119.   

## Description

This pull request includes several updates and improvements:
1. **Update SPA Dependencies** - Updated the Single Page Application (SPA) dependencies to ensure compatibility with the latest features and security fixes.
2. **Update DNN Dependency to version 9.11.00** - Updated the DNN dependency to version 9.11.00 to ensure compatibility with the latest version of DNN and to address any potential issues related to older versions.
3. **Add an Unpublished Pages Toggle Switch** -
- ​​Removed the word "filter" from the existing **Metadata** toggle switch label.
- Added a new switch labeled **Unpublished**, which allows users to toggle unpublished pages.
- The default state of the **Unpublished** toggle switch is set to **Off**.
- Only one of the toggles (**Metadata** or **Unpublished**) can be active at a time. When one switch is turned on, the other will automatically be turned off, ensuring that only one filter condition is applied at a given time.
4. **Move module project to Modules folder** - The module project has been moved to the correct folder structure. The `Upendo.Modules.DnnPageManager` folder is now nested inside the `Modules` folder, ensuring proper organization and alignment with the project structure.
- The previous solution location was disorganized and did not follow the expected folder structure.
- After this move, all build, compilation, and packaging tasks have been tested to ensure everything works as expected.
5. **Update build files** - Updated build files based on the latest version of the UpendoDNN generator.
6. **Fix packaging of generated symbols** - Fixed issues related to packaging of generated symbols, ensuring they are properly included in the final build result.
7. **Debug build for correct symbol packaging**:
- **Bug description**: The current build process creates a symbol package, but it is not usable. The symbol package is created incorrectly, causing issues with installation and uninstallation.
- **Tasks addressed**:
1. **Remove packages from repository**: Ensured that packages are not checked into the repository and removed from the codebase.
2. **Save to correct folder**: Configured the build process to save symbol packages to the `/Website/Install/Module/` folder instead of the local project folder to ensure correct packaging and access.

## How Has This Been Tested?

The changes were tested through the following:
- **SPA Dependency Update**: Verified that the SPA functionalities work as expected after updating the dependencies.
- **DNN Dependency Update**: Confirmed that the project runs smoothly with DNN 9.11.00, and no breaking changes were encountered.
- **Unpublished Pages Toggle**:
  - Verified that the **Unpublished** toggle appears next to the **Metadata** toggle.
  - Ensured that only one of the toggles (**Metadata** or **Unpublished**) can be turned on at a time. Switching one toggle off automatically turns the other on if necessary.
- **Move Module Project into the Modules Folder**: 
  - Verified that the `Upendo.Modules.DnnPageManager` folder is correctly nested inside the `Modules` folder.
  - Confirmed that all build, compile, and packaging tasks work as expected after the move.
- **UpendoDNN Generator Build Files**: Verified the updated build files work as expected and that the project builds without errors.
- **Symbols Packaging Fix**: 
  - Verified that the symbols packages are now created correctly and saved in the `/Website/Install/Module/` folder.
  - Confirmed that the packages are no longer checked in the repository and that the `.gitignore` file properly ignores them.

## Screenshots (if appropriate):
image

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: References/DNN/09.11.00/DotNetNuke.DependencyInjection.deps.json [hunk 1]
```diff
@@ -74,12 +74,12 @@
...
       "sha512": ""
     },
-    "Microsoft.Build.Tasks.Git/1.0.0": {
+    "Microsoft.Build.Tasks.Git/1.1.1": {
       "type": "package",
       "serviceable": true,
-      "sha512": "sha512-z2fpmmt+1Jfl+ZnBki9nSP08S1/tbEOxFdsK1rSR+LBehIJz1Xv9/6qOOoGNqlwnAGGVGis1Oj6S8Kt9COEYlQ==",
-      "path": "microsoft.build.tasks.git/1.0.0",
-      "hashPath": "microsoft.build.tasks.git.1.0.0.nupkg.sha512"
+      "sha512": "sha512-AT3HlgTjsqHnWpBHSNeR0KxbLZD7bztlZVj7I8vgeYG9SYqbeFGh0TM/KVtC6fg53nrWHl3VfZFvb5BiQFcY6Q==",
```
Comment (WillStrohl) [inline References/DNN/09.11.00/DotNetNuke.DependencyInjection.deps.json:80]: @ArielBlanco1990 Should this be committed? 🤔

File: References/DNN/09.11.00/DotNetNuke.DependencyInjection.deps.json [hunk 2]
```diff
@@ -102,19 +102,19 @@
...
       "hashPath": "microsoft.netcore.platforms.1.1.0.nupkg.sha512"
     },
-    "Microsoft.SourceLink.Common/1.0.0": {
+    "Microsoft.SourceLink.Common/1.1.1": {
       "type": "package",
       "serviceable": true,
-      "sha512": "sha512-G8DuQY8/DK5NN+3jm5wcMcd9QYD90UV7MiLmdljSJixi3U/vNaeBKmmXUqI4DJCOeWizIUEh4ALhSt58mR+5eg==",
-      "path": "microsoft.sourcelink.common/1.0.0",
-      "hashPath": "microsoft.sourcelink.common.1.0.0.nupkg.sha512"
+      "sha512": "sha512-WMcGpWKrmJmzrNeuaEb23bEMnbtR/vLmvZtkAP5qWu7vQsY59GqfRJd65sFpBszbd2k/bQ8cs8eWawQKAabkVg==",
```
Comment (WillStrohl) [inline References/DNN/09.11.00/DotNetNuke.DependencyInjection.deps.json:108]: @ArielBlanco1990 Should this be committed? 🤔

File: References/DNN/09.11.00/DotNetNuke.DependencyInjection.deps.json [hunk 3]
```diff
@@ -102,19 +102,19 @@
...
+      "hashPath": "microsoft.sourcelink.common.1.1.1.nupkg.sha512"
     },
-    "Microsoft.SourceLink.GitHub/1.0.0": {
+    "Microsoft.SourceLink.GitHub/1.1.1": {
       "type": "package",
       "serviceable": true,
-      "sha512": "sha512-aZyGyGg2nFSxix+xMkPmlmZSsnGQ3w+mIG23LTxJZHN+GPwTQ5FpPgDo7RMOq+Kcf5D4hFWfXkGhoGstawX13Q==",
-      "path": "microsoft.sourcelink.github/1.0.0",
-      "hashPath": "microsoft.sourcelink.github.1.0.0.nupkg.sha512"
+      "sha512": "sha512-IaJGnOv/M7UQjRJks7B6p7pbPnOwisYGOIzqCz5ilGFTApZ3ktOR+6zJ12ZRPInulBmdAf1SrGdDG2MU8g6XTw==",
```
Comment (WillStrohl) [inline References/DNN/09.11.00/DotNetNuke.DependencyInjection.deps.json:115]: @ArielBlanco1990 Should this be committed? 🤔

**Review by WillStrohl — CHANGES_REQUESTED** [review]
There are some minor changes required.

File: Build/ModulePackage.targets.bak [hunk 4]
```diff
@@ -0,0 +1,148 @@
+<?xml version="1.0" encoding="utf-8"?>
```
[the remark this answers is not in the document]
Comment (ArielBlanco1990) [reply Build/ModulePackage.targets.bak:1]: @WillStrohl It seems that it was created automatically when I updated the file, I will delete it.

File: References/DNN/09.11.00/DotNetNuke.Abstractions.deps.json [hunk 5]
```diff
@@ -40,12 +40,12 @@
...
       "sha512": ""
     },
-    "Microsoft.Build.Tasks.Git/1.0.0": {
+    "Microsoft.Build.Tasks.Git/1.1.1": {
       "type": "package",
       "serviceable": true,
-      "sha512": "sha512-z2fpmmt+1Jfl+ZnBki9nSP08S1/tbEOxFdsK1rSR+LBehIJz1Xv9/6qOOoGNqlwnAGGVGis1Oj6S8Kt9COEYlQ==",
-      "path": "microsoft.build.tasks.git/1.0.0",
-      "hashPath": "microsoft.build.tasks.git.1.0.0.nupkg.sha512"
+      "sha512": "sha512-AT3HlgTjsqHnWpBHSNeR0KxbLZD7bztlZVj7I8vgeYG9SYqbeFGh0TM/KVtC6fg53nrWHl3VfZFvb5BiQFcY6Q==",
```
[the remark this answers is not in the document]
Comment (ArielBlanco1990) [reply References/DNN/09.11.00/DotNetNuke.Abstractions.deps.json:46]: @WillStrohl I don't think it should be in the commit, I'll delete it, it was inside the DNN 09.11.00 folder and I didn't notice it.

File: References/DNN/09.11.00/DotNetNuke.Abstractions.deps.json [hunk 6]
```diff
@@ -54,19 +54,19 @@
...
       "hashPath": "microsoft.netcore.platforms.1.1.0.nupkg.sha512"
     },
-    "Microsoft.SourceLink.Common/1.0.0": {
+    "Microsoft.SourceLink.Common/1.1.1": {
       "type": "package",
       "serviceable": true,
-      "sha512": "sha512-G8DuQY8/DK5NN+3jm5wcMcd9QYD90UV7MiLmdljSJixi3U/vNaeBKmmXUqI4DJCOeWizIUEh4ALhSt58mR+5eg==",
-      "path": "microsoft.sourcelink.common/1.0.0",
-      "hashPath": "microsoft.sourcelink.common.1.0.0.nupkg.sha512"
+      "sha512": "sha512-WMcGpWKrmJmzrNeuaEb23bEMnbtR/vLmvZtkAP5qWu7vQsY59GqfRJd65sFpBszbd2k/bQ8cs8eWawQKAabkVg==",
```
[the remark this answers is not in the document]
Comment (ArielBlanco1990) [reply References/DNN/09.11.00/DotNetNuke.Abstractions.deps.json:60]: @WillStrohl I don't think it should be in the commit, I'll delete it, it was inside the DNN 09.11.00 folder and I didn't notice it.

File: References/DNN/09.11.00/DotNetNuke.Abstractions.deps.json [hunk 7]
```diff
@@ -54,19 +54,19 @@
...
+      "hashPath": "microsoft.sourcelink.common.1.1.1.nupkg.sha512"
     },
-    "Microsoft.SourceLink.GitHub/1.0.0": {
+    "Microsoft.SourceLink.GitHub/1.1.1": {
       "type": "package",
       "serviceable": true,
-      "sha512": "sha512-aZyGyGg2nFSxix+xMkPmlmZSsnGQ3w+mIG23LTxJZHN+GPwTQ5FpPgDo7RMOq+Kcf5D4hFWfXkGhoGstawX13Q==",
-      "path": "microsoft.sourcelink.github/1.0.0",
-      "hashPath": "microsoft.sourcelink.github.1.0.0.nupkg.sha512"
+      "sha512": "sha512-IaJGnOv/M7UQjRJks7B6p7pbPnOwisYGOIzqCz5ilGFTApZ3ktOR+6zJ12ZRPInulBmdAf1SrGdDG2MU8g6XTw==",
```
[the remark this answers is not in the document]
Comment (ArielBlanco1990) [reply References/DNN/09.11.00/DotNetNuke.Abstractions.deps.json:67]: @WillStrohl I don't think it should be in the commit, I'll delete it, it was inside the DNN 09.11.00 folder and I didn't notice it.

File: References/DNN/09.11.00/DotNetNuke.DependencyInjection.deps.json (hunk 1 above)
Comment (ArielBlanco1990) [reply References/DNN/09.11.00/DotNetNuke.DependencyInjection.deps.json:80]: @WillStrohl I don't think it should be in the commit, I'll delete it, it was inside the DNN 09.11.00 folder and I didn't notice it.

File: References/DNN/09.11.00/DotNetNuke.DependencyInjection.deps.json (hunk 2 above)
Comment (ArielBlanco1990) [reply References/DNN/09.11.00/DotNetNuke.DependencyInjection.deps.json:108]: @WillStrohl I don't think it should be in the commit, I'll delete it, it was inside the DNN 09.11.00 folder and I didn't notice it.

File: References/DNN/09.11.00/DotNetNuke.DependencyInjection.deps.json (hunk 3 above)
Comment (ArielBlanco1990) [reply References/DNN/09.11.00/DotNetNuke.DependencyInjection.deps.json:115]: @WillStrohl I don't think it should be in the commit, I'll delete it, it was inside the DNN 09.11.00 folder and I didn't notice it.

**Comment by ArielBlanco1990** [comment]
@WillStrohl I have resolved each of the details mentioned

File: References/DNN/09.11.00/DotNetNuke.Abstractions.deps.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (WillStrohl) [reply References/DNN/09.11.00/DotNetNuke.Abstractions.deps.json:60]: @ArielBlanco1990 I don't see a commit that addresses this.  🤔

File: References/DNN/09.11.00/DotNetNuke.Abstractions.deps.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (WillStrohl) [reply References/DNN/09.11.00/DotNetNuke.Abstractions.deps.json:67]: @ArielBlanco1990 I don't see a commit that addresses this.  🤔

File: References/DNN/09.11.00/DotNetNuke.DependencyInjection.deps.json (hunk 1 above)
Comment (WillStrohl) [reply References/DNN/09.11.00/DotNetNuke.DependencyInjection.deps.json:80]: @ArielBlanco1990 I don't see a commit that addresses this.  🤔

File: References/DNN/09.11.00/DotNetNuke.DependencyInjection.deps.json (hunk 2 above)
Comment (WillStrohl) [reply References/DNN/09.11.00/DotNetNuke.DependencyInjection.deps.json:108]: @ArielBlanco1990 I don't see a commit that addresses this.  🤔

File: References/DNN/09.11.00/DotNetNuke.DependencyInjection.deps.json (hunk 3 above)
Comment (WillStrohl) [reply References/DNN/09.11.00/DotNetNuke.DependencyInjection.deps.json:115]: @ArielBlanco1990 I don't see a commit that addresses this.  🤔

**Review by WillStrohl — CHANGES_REQUESTED** [review]
@ArielBlanco1990 A few things were missed and still need to be addressed.

**Comment by ArielBlanco1990** [comment]
@WillStrohl I missed those comments when I went back to check the comments again, I thought I had deleted them, it was my mistake. They are now removed in the commit https://github.com/UpendoVentures/Upendo-Dnn-PageManager/pull/121/commits/6b92b87c561d9338eed067b4846a6d4ba5b88be5.

**Review by WillStrohl — APPROVED** [review]
Looks good!

**Pull request merged**